tezos-sig-gating-contracts/scripts/tests/gatednftminter_nodispatch.spec.ts

```diff
@@ -0,0 +1,719 @@
+import { expect } from "chai";
+import { deployNFTMinterExtNoDispatch } from "../fixtures/fixtureGatedNftMinter_nodispatch";
+import { InMemorySigner } from "@taquito/signer";
+import {
+  MichelsonMap,
+  TezosToolkit,
+  TezosOperationError,
+} from "@taquito/taquito";
+import {
+  convert_timestamp,
+  convert_key,
+  convert_nat,
+  convert_string,
+  convert_address,
+  convert_chain_id,
+  convert_mint,
+} from "../utils/convert";
+import {
+  EdSignature,
+  TezosTxAuthData,
+  TezosTxCalldata,
+} from "../utils/schemas";
+
+const RPC_ENDPOINT = "http://localhost:20000/";
+
+const Tezos = new TezosToolkit(RPC_ENDPOINT);
+import { RpcClient } from "@taquito/rpc";
+const client = new RpcClient(RPC_ENDPOINT); //, 'NetXnofnLBXBoxo');
+
+const nexeraSigner = new InMemorySigner(
+  "edsk3RFfvaFaxbHx8BMtEW1rKQcPtDML3LXjNqMNLCzC3wLC1bWbAt"
+); // signer private key
+
+const createKeccakHash = require("keccak");
+
+function keccak256(data: string) {
+  return createKeccakHash("keccak256").update(data, "hex").digest("hex");
+}
+
+function computePayloadHash(payload: TezosTxAuthData) {
+  const nonce_string = payload.nonce.toString();
+  const expiration_string = payload.blockExpiration.toString();
+
+  const chain_id_bytes = convert_chain_id(payload.chainID);
+  const user_bytes = convert_address(payload.userAddress);
+  const functioncall_contract_bytes = convert_address(payload.contractAddress);
+  const functioncall_name_bytes = convert_string(payload.functionCallName);
+  const functionCallArgsBytes = payload.functionCallArgs;
+  const nonce_bytes = convert_nat(nonce_string);
+  const expiration_bytes = convert_nat(expiration_string);
+  const key_bytes = convert_key(payload.signerPublicKey);
+  const payload_bytes =
+    key_bytes +
+    chain_id_bytes +
+    user_bytes +
+    nonce_bytes +
+    expiration_bytes +
+    functioncall_contract_bytes +
+    functioncall_name_bytes +
+    functionCallArgsBytes;
+  const payload_hash = keccak256(payload_bytes);
+  return payload_hash;
+}
+
+function buildTxInputFromTxAuthData(
```

Comment:
maybe it would be better to put these helper functions under ../utils/buildTxCalldata and give them different names (buildTxCalldataNoContractAddress for example)

Reply:
agreed